### ISSUE TITLE
Feature: Reorganize design system to clarify responsibilities of different components

### ISSUE THREAD

**Issue by brandonroberts** [open-sauced/insights#791]
### Type of feature

🧑‍💻 Refactor

### Current behavior

Currently, our design system is built using the principles of the Atomic Design system, which consists of atoms, molecules, organisms, and templates/pages. Over time we have diverged away from this somewhat, with anything above atoms potentially not remaining stateless and adhering to the standard.  This also presents a barrier to new contributors of where to put which components without having to watch the content on the design system pattern first.

### Suggested solution

This proposed change:
- Renames `atoms` to `primitives` which still remain the most basic building blocks of the design system
- Groups all other components under a `components`  folder.
  - We still want to keep these components relatively stateless, but it does not completely prevent them from having some internal state
- Refactors existing imports to use the new import paths
- Updates the documentation about the design system in general and where primitives and components should live.
- Also updates the Storybook stories

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

**Comment by TAKANOME-DEV** [comment]
@brandonroberts if I understand correctly, the folder structure should be like this:
```bash
├── components
│   ├── primitives
│   │   ├── Button
│   │   │   ├── button.tsx
│   │   ├── components
│   │   │   ├── Header
│   │   │   ├── Dashboard
│   │   │   ├── User Profile
│   │   │   ├── Charts
│   │   │   ├── ...
├── stories
│   ├── primitives
│   │   ├── button.stories.tsx
│   │   ├── components
│   │   │   ├── Header
│   │   │   ├── Dashboard
│   │   │   ├── User Profile
│   │   │   ├── Charts
│   │   │   ├── ...
```

**Comment by brandonroberts** [comment]
@TAKANOME-DEV yes, that's true. As a side note, we're not ready to implement this change yet. If you would like to pick up an issue, I'd check Papercuts milestone first.

**Comment by TAKANOME-DEV** [comment]
> As a side note, we're not ready to implement this change yet. 
>
Yeah, I thought so. Thanks
